DOC: Fixed docstring for mode()

MarianD opened the pull request:
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

MarianD committed:
DOC: Fixed docstring for mode()

MarianD committed:
DOC: Changed docstring for mode()

 - Explanation of mode() function

MarcoGorelli commented on pandas/core/series.py:
```diff
@@ -1786,7 +1786,10 @@ def count(self, level=None):
     def mode(self, dropna=True) -> "Series":
         """
-        Return the mode(s) of the dataset.
+        Return the mode(s) of the series.
```
I think "Series" is usually capitalised in the docs - other than that, lgtm, thanks
```suggestion
        Return the mode(s) of the Series.
```

MarianD committed:
Update pandas/core/series.py

Co-authored-by: Marco Gorelli <m.e.gorelli@gmail.com>

dsaxton commented on pandas/core/series.py:
```diff
@@ -1786,7 +1786,10 @@ def count(self, level=None):
+        Return the mode(s) of the Series.
+
+        The mode is the value that appears most often.
+        It can be multiple values.
```
Can this fit on one line? Otherwise LGTM.

MarianD replied:
The first line of a docstring is limited to 55 characters. Other 2 lines in my suggestion will be displayed in the documentation anyway as 1 line.

MarianD committed:
DOC: Changes in docstring for mode()

 - Fixed 1st line of docstring
 - Added explanation for mode() function
 - Explanation for mode(): joined 2 lines into 1

MarianD committed:
Update pandas/core/series.py

Co-authored-by: Daniel Saxton <2658661+dsaxton@users.noreply.github.com>
